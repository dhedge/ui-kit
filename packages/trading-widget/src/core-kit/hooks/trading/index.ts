export { useTradingResultHandling } from './use-trading-result-handling'
export { useHandlePoolSwapInfo } from './use-handle-pool-swap-info'
export { useTradingSettleHandler } from './use-trading-settle-handler'
export { useTradingPriceDiff } from './use-trading-price-diff'
export { useAssetPrice } from './use-asset-price'
export { useRawAssetPrice } from './use-raw-asset-price'
export { useIsTradingEnabled } from './use-is-trading-enabled'
export { useExchangeRate } from './use-exchange-rate'
export { useHandleTrade } from './use-handle-trade'
export { useMinReceiveText } from './use-min-receive-text'
export { useMaxSlippagePlaceholder } from './use-max-slippage-placeholder'
export { useSynthetixV3OraclesUpdate } from './synthetix-v3/use-synthetix-v3-oracles-update'
export { useDepositProjectedEarnings } from './projected-earnings/use-deposit-projected-earnings'
export { useProjectedEarningsCore } from './projected-earnings/use-projected-earnings-core'
export { useIsEasySwapperTrading } from './use-is-easy-swapper-trading'
export { useAvailableWithdrawLiquidity } from './synthetix-v3/use-available-withdraw-liquidity'
