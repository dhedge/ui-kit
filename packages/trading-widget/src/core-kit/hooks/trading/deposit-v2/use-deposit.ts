import { useCallback } from 'react'

import {
  useReceiveTokenInput,
  useTradingPanelPoolConfig,
  useTradingPanelTransactions,
} from 'core-kit/hooks/state'
import { useTradingSettleHandler } from 'core-kit/hooks/trading'
import { useContractFunction } from 'core-kit/hooks/web3'

import type { ContractActionFunc } from 'core-kit/types/web3.types'

import { useVaultDepositParams } from './use-vault-deposit-params'
import { useVaultDepositTransactionArguments } from './use-vault-deposit-transaction-arguments'

const action = 'deposit'

export const useDeposit = (): ContractActionFunc => {
  const { chainId } = useTradingPanelPoolConfig()
  const [receiveToken] = useReceiveTokenInput()
  const updatePendingTransactions = useTradingPanelTransactions()[1]
  const { depositMethod, vaultDepositTokenAddress } = useVaultDepositParams()
  const txArgs = useVaultDepositTransactionArguments({
    depositMethod,
    vaultDepositTokenAddress,
  })

  const onSettled = useTradingSettleHandler(action)

  const { send } = useContractFunction({
    contractId: 'easySwapperV2',
    functionName: depositMethod,
    onSettled,
  })

  return useCallback(async () => {
    updatePendingTransactions({
      type: 'add',
      action,
      symbol: receiveToken.symbol,
      chainId,
    })

    console.log('Function Name:', depositMethod)
    console.log('Transaction Arguments:', txArgs)
    return send(...txArgs)
  }, [
    updatePendingTransactions,
    chainId,
    receiveToken.symbol,
    send,
    txArgs,
    depositMethod,
  ])
}
