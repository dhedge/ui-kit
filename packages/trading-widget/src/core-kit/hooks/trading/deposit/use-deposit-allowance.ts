import BigNumber from 'bignumber.js'

import { useEffect } from 'react'

import { AddressZero } from 'core-kit/const'
import {
  useSendTokenInput,
  useTradingPanelApprovingStatus,
  useTradingPanelPoolConfig,
} from 'core-kit/hooks/state'
import { useApprove, useCanSpend } from 'core-kit/hooks/trading/allowance'
import { useAccount } from 'core-kit/hooks/web3'
import { getContractAddressById, isNativeToken } from 'core-kit/utils'

import { useIsEasySwapperTrading } from '../use-is-easy-swapper-trading'

export const useDepositAllowance = () => {
  const { account } = useAccount()
  const poolConfig = useTradingPanelPoolConfig()
  const [sendToken] = useSendTokenInput()
  const updateApprovingStatus = useTradingPanelApprovingStatus()[1]
  const isEasySwapperTrading = useIsEasySwapperTrading()

  const rawDepositAmount = new BigNumber(sendToken.value || '0')
    .shiftedBy(sendToken.decimals)
    .toFixed(0, BigNumber.ROUND_UP)

  const spenderAddress = isEasySwapperTrading
    ? getContractAddressById('easySwapper', poolConfig.chainId)
    : poolConfig.address

  const canSpend = useCanSpend({
    tokenAddress: sendToken.address,
    ownerAddress: account ?? AddressZero,
    spenderAddress,
    rawAmountToSpend: rawDepositAmount,
    chainId: poolConfig.chainId,
    skip:
      isNativeToken(sendToken.symbol, poolConfig.chainId) ||
      rawDepositAmount === '0',
  })
  const approve = useApprove({
    token: sendToken,
    rawTokenAmount: rawDepositAmount,
    spenderAddress,
  })

  useEffect(() => {
    updateApprovingStatus(canSpend ? 'success' : undefined)
  }, [updateApprovingStatus, canSpend])

  return { approve, canSpend }
}
