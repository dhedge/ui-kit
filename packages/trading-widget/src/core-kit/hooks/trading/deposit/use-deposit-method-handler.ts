import BigNumber from 'bignumber.js'
import { useCallback, useMemo } from 'react'

import { usePoolDynamicContractData } from 'core-kit/hooks/pool'
import { usePoolStatic } from 'core-kit/hooks/pool/multicall'
import {
  useTradingPanelDepositMethod,
  useTradingPanelPoolConfig,
} from 'core-kit/hooks/state'
import { useIsPoolManagerAccount } from 'core-kit/hooks/user'
import type { DepositMethodName } from 'core-kit/types/trading-panel.types'

export const useDepositMethodHandler = (): [
  DepositMethodName,
  (method: DepositMethodName) => void,
  boolean,
] => {
  const { address, chainId } = useTradingPanelPoolConfig()
  const [depositMethod, setDepositMethod] = useTradingPanelDepositMethod()
  const isPoolManagerAccount = useIsPoolManagerAccount()
  const { data: { easySwapperAllowedPools: isPoolAllowed = false } = {} } =
    usePoolStatic({
      address,
      chainId,
    })
  const { entryFee = '0' } = usePoolDynamicContractData({
    address,
    chainId,
  })

  const entryFeeBN = new BigNumber(entryFee)
  const hasDepositOptions =
    (entryFeeBN.isNaN() || entryFeeBN.isZero()) &&
    isPoolAllowed &&
    !isPoolManagerAccount

  const setPoolDepositMethod = useCallback(
    (method: DepositMethodName) => {
      setDepositMethod({
        address,
        method,
      })
    },
    [setDepositMethod, address],
  )

  return useMemo(
    () => [depositMethod, setPoolDepositMethod, hasDepositOptions],
    [depositMethod, setPoolDepositMethod, hasDepositOptions],
  )
}
