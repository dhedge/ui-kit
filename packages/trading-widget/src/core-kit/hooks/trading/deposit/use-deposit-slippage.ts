import { useEffect } from 'react'

import {
  useIsDepositTradingPanelType,
  useSendTokenInput,
  useTradingPanelSettings,
} from 'core-kit/hooks/state'
import { useTradingPriceDiff } from 'core-kit/hooks/trading'
import { useDebounce } from 'core-kit/hooks/utils'

export const useDepositSlippage = (receiveAssetInputValue: string) => {
  const isDeposit = useIsDepositTradingPanelType()
  const [sendToken] = useSendTokenInput()
  const updateSettings = useTradingPanelSettings()[1]
  const tradingPriceDiff = useTradingPriceDiff({
    sendAssetValue: sendToken.value,
    sendAssetAddress: sendToken.address,
    receiveAssetValue: receiveAssetInputValue,
  })
  const priceDiffDebounced = useDebounce(tradingPriceDiff, 100)

  useEffect(() => {
    if (isDeposit) {
      updateSettings({ minSlippage: Math.abs(priceDiffDebounced) })
    }
  }, [updateSettings, isDeposit, priceDiffDebounced])
}
