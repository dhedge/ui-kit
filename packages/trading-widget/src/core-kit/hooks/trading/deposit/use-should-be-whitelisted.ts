import {
  useCheckWhitelist,
  usePoolDynamicContractData,
} from 'core-kit/hooks/pool'
import { useTradingPanelPoolConfig } from 'core-kit/hooks/state'

export const useShouldBeWhitelisted = () => {
  const { chainId, address, deprecated } = useTradingPanelPoolConfig()
  const { isPrivateVault = false } = usePoolDynamicContractData({
    address,
    chainId,
  })

  const shouldBeWhitelisted = isPrivateVault || deprecated
  const isAccountWhitelisted = useCheckWhitelist({
    address,
    chainId,
  })

  return { shouldBeWhitelisted, isAccountWhitelisted }
}
