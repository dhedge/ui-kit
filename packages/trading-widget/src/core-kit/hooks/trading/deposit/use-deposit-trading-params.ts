import BigNumber from 'bignumber.js'
import { useMemo } from 'react'

import { usePoolTokenPrice } from 'core-kit/hooks/pool'
import {
  useReceiveTokenInput,
  useSendTokenInput,
  useTradingPanelPoolConfig,
  useTradingPanelSettings,
} from 'core-kit/hooks/state'
import { usePoolDepositAssetAddress } from 'core-kit/hooks/trading/deposit'
import { useAssetPrice } from 'core-kit/hooks/trading/use-asset-price'
import type { TradingParams } from 'core-kit/types/trading.types'

export const useDepositTradingParams = (): TradingParams => {
  const { address, chainId } = useTradingPanelPoolConfig()
  const [{ slippage }] = useTradingPanelSettings()
  const isAutoSlippage = slippage === 'auto'

  const [sendToken] = useSendTokenInput()
  const sendTokenPrice =
    useAssetPrice({
      address: sendToken.address,
      chainId,
      disabled: isAutoSlippage,
    }) ?? ''

  const [receiveToken] = useReceiveTokenInput()
  const receiveAssetAddress = receiveToken.address
  const receiveAssetPrice =
    usePoolTokenPrice({ address, chainId, disabled: isAutoSlippage }) ?? ''

  const poolDepositAssetAddress = usePoolDepositAssetAddress({
    investAssetAddress: sendToken.address,
    symbol: sendToken.symbol,
    productPoolAddress: address,
    chainId,
  })
  const manualSlippageReceiveAssetAmount = new BigNumber(sendToken.value || '0')
    .multipliedBy(sendTokenPrice)
    .dividedBy(receiveAssetPrice)
    .toFixed()
  const receiveAssetAmount = isAutoSlippage
    ? receiveToken.value
    : manualSlippageReceiveAssetAmount

  return useMemo<TradingParams>(
    () => ({
      sendAssetAddress: sendToken.address,
      fromTokenAmount: new BigNumber(sendToken.value || '0'),
      receiveAssetAddress,
      receiveAssetAmount,
      poolDepositAddress: poolDepositAssetAddress,
    }),
    [
      sendToken.address,
      sendToken.value,
      receiveAssetAddress,
      receiveAssetAmount,
      poolDepositAssetAddress,
    ],
  )
}
