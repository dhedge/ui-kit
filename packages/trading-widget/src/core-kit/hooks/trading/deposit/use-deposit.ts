import { useCallback, useMemo } from 'react'

import {
  useReceiveTokenInput,
  useSendTokenInput,
  useTradingPanelDepositMethod,
  useTradingPanelPoolConfig,
  useTradingPanelSettings,
  useTradingPanelTransactions,
} from 'core-kit/hooks/state'
import {
  useTradingParams,
  useTradingSettleHandler,
} from 'core-kit/hooks/trading'
import { useDepositSlippage } from 'core-kit/hooks/trading/deposit'
import { useContractFunction } from 'core-kit/hooks/web3'

import {
  BuyingWithEasyswapperArgs,
  BuyingWithNativeAssetArgs,
  BuyingWithPoolLogicArgs,
} from 'core-kit/models'
import type { ContractActionFunc } from 'core-kit/types/web3.types'
import {
  getOrderedTxArgs,
  isNativeToken,
  logTransactionArguments,
} from 'core-kit/utils'

import { useIsEasySwapperTrading } from '../use-is-easy-swapper-trading'
import { useConfigContextParams } from 'trading-widget/providers/config-provider'

const action = 'deposit'

export const useDeposit = (): ContractActionFunc => {
  const poolConfig = useTradingPanelPoolConfig()
  const [depositMethod] = useTradingPanelDepositMethod()
  const [sendToken] = useSendTokenInput()
  const [receiveToken] = useReceiveTokenInput()
  const updatePendingTransactions = useTradingPanelTransactions()[1]
  const [{ slippage }] = useTradingPanelSettings()
  const isEasySwapperDeposit = useIsEasySwapperTrading()
  const {
    poolDepositAddress,
    receiveAssetAddress,
    sendAssetAddress,
    receiveAssetInputValue,
    fromTokenAmount,
  } = useTradingParams()
  const { defaultDepositSlippage } = useConfigContextParams()

  useDepositSlippage(receiveAssetInputValue)

  const isDepositNative =
    poolConfig.chainId && isEasySwapperDeposit
      ? isNativeToken(sendToken.symbol, poolConfig.chainId)
      : false
  const functionName = isDepositNative
    ? depositMethod === 'deposit'
      ? 'depositNative'
      : 'depositNativeWithCustomCooldown'
    : depositMethod

  const onSettled = useTradingSettleHandler(action)
  const txArgs = useMemo(() => {
    const depositAddress = poolDepositAddress ?? sendAssetAddress
    const depositArgs = {
      receiveAssetAddress,
      fromTokenAmount: fromTokenAmount.shiftedBy(sendToken.decimals).toFixed(0),
      poolDepositAddress: depositAddress,
      receiveAssetInputValue,
    }

    if (!isEasySwapperDeposit) {
      return new BuyingWithPoolLogicArgs({
        ...depositArgs,
        fromTokenAddress: sendAssetAddress,
      })
    }

    if (isDepositNative) {
      return new BuyingWithNativeAssetArgs(depositArgs)
    }

    return new BuyingWithEasyswapperArgs({
      ...depositArgs,
      sendAssetAddress,
    })
  }, [
    fromTokenAmount,
    isDepositNative,
    isEasySwapperDeposit,
    poolDepositAddress,
    receiveAssetAddress,
    receiveAssetInputValue,
    sendAssetAddress,
    sendToken.decimals,
  ])
  const { send } = useContractFunction({
    contractId: isEasySwapperDeposit ? 'easySwapper' : 'poolLogic',
    dynamicContractAddress: isEasySwapperDeposit
      ? undefined
      : receiveToken.address,
    functionName,
    onSettled,
  })

  return useCallback(async () => {
    updatePendingTransactions({
      type: 'add',
      action,
      symbol: receiveToken.symbol,
      chainId: poolConfig.chainId,
    })

    logTransactionArguments(txArgs)
    const args: unknown[] = getOrderedTxArgs(
      txArgs,
      slippage === 'auto' ? defaultDepositSlippage : slippage,
    )
    if (isDepositNative) {
      args.push({ value: BigInt(txArgs.fromTokenAmount) })
    }
    return send(...args)
  }, [
    updatePendingTransactions,
    isDepositNative,
    poolConfig.chainId,
    receiveToken.symbol,
    send,
    txArgs,
    slippage,
    defaultDepositSlippage,
  ])
}
