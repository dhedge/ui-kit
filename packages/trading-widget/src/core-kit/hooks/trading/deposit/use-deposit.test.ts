import { act } from '@testing-library/react'
import BigNumber from 'bignumber.js'

import { AddressZero, BRIDGED_USDC_OPTIMISM, optimism } from 'core-kit/const'
import * as stateHooks from 'core-kit/hooks/state'

import {
  useTradingParams,
  useTradingSettleHandler,
} from 'core-kit/hooks/trading'
import { useDepositSlippage } from 'core-kit/hooks/trading/deposit'
import { useContractFunction } from 'core-kit/hooks/web3'
import type { Address, DynamicTradingToken } from 'core-kit/types'
import { renderHook } from 'tests/test-utils'

import { useDeposit } from './use-deposit'

vi.mock('core-kit/hooks/state', () => ({
  useReceiveTokenInput: vi.fn(),
  useSendTokenInput: vi.fn(),
  useTradingPanelDepositMethod: vi.fn(),
  useTradingPanelPoolConfig: vi.fn(),
  useTradingPanelSettings: vi.fn(),
  useTradingPanelTransactions: vi.fn(),
}))
vi.mock('core-kit/hooks/trading', () => ({
  useTradingSettleHandler: vi.fn(),
  useTradingParams: vi.fn(),
}))
vi.mock('core-kit/hooks/trading/deposit', () => ({
  useDepositSlippage: vi.fn(),
}))
vi.mock('core-kit/hooks/web3', () => ({
  useContractFunction: vi.fn(),
}))
const { logTransactionArguments } = vi.hoisted(() => ({
  logTransactionArguments: vi.fn(),
}))
vi.mock('core-kit/utils', async () => {
  const actual =
    await vi.importActual<Record<string, unknown>>('core-kit/utils')
  return {
    ...actual,
    logTransactionArguments,
  }
})
describe('useDeposit', () => {
  const chainId = optimism.id
  const updatePendingTransactionsMock = vi.fn()
  const onSettledMock = vi.fn()
  const sendMock = vi.fn()

  beforeEach(() => {
    vi.mocked(stateHooks.useTradingPanelPoolConfig).mockImplementation(
      () =>
        ({ chainId }) as ReturnType<
          typeof stateHooks.useTradingPanelPoolConfig
        >,
    )
    vi.mocked(stateHooks.useTradingPanelTransactions).mockImplementation(() => [
      [],
      updatePendingTransactionsMock,
    ])
    vi.mocked(useTradingSettleHandler).mockImplementation(() => onSettledMock)
    vi.mocked(useContractFunction).mockImplementation(
      () =>
        ({
          send: sendMock,
        }) as unknown as ReturnType<typeof useContractFunction>,
    )
  })

  afterEach(() => {
    vi.restoreAllMocks()
  })

  it('should process deposits with auto slippage and not native token', async () => {
    const depositMethod = 'deposit'
    const sendToken = { ...BRIDGED_USDC_OPTIMISM, value: '5', isLoading: false }
    const receiveToken = {
      symbol: 'USDy',
      address: '0x123',
      value: '10',
    } as unknown as DynamicTradingToken
    const slippage = 'auto'
    const tradingParams = {
      poolDepositAddress: '0x111' as Address,
      receiveAssetAddress: receiveToken.address,
      sendAssetAddress: sendToken.address,
      receiveAssetInputValue: receiveToken.value,
      fromTokenAmount: new BigNumber(sendToken.value),
    }
    vi.mocked(stateHooks.useTradingPanelDepositMethod).mockImplementation(
      () => [depositMethod, vi.fn()],
    )
    vi.mocked(stateHooks.useSendTokenInput).mockImplementation(() => [
      sendToken,
      vi.fn(),
    ])
    vi.mocked(stateHooks.useReceiveTokenInput).mockImplementation(() => [
      receiveToken,
      vi.fn(),
    ])
    vi.mocked(stateHooks.useTradingPanelSettings).mockImplementation(
      () =>
        [
          {
            slippage,
          },
          vi.fn,
        ] as unknown as ReturnType<typeof stateHooks.useTradingPanelSettings>,
    )
    vi.mocked(useTradingParams).mockImplementation(() => tradingParams)

    const { result } = renderHook(() => useDeposit())

    expect(useDepositSlippage).toHaveBeenCalledTimes(1)
    expect(useDepositSlippage).toHaveBeenCalledWith(
      tradingParams.receiveAssetInputValue,
    )
    expect(useTradingSettleHandler).toHaveBeenCalledTimes(1)
    expect(useTradingSettleHandler).toHaveBeenCalledWith('deposit')

    expect(useContractFunction).toHaveBeenCalledTimes(1)
    expect(useContractFunction).toHaveBeenCalledWith({
      contractId: 'easySwapper',
      functionName: depositMethod,
      onSettled: onSettledMock,
    })

    await act(async () => await result.current())

    expect(updatePendingTransactionsMock).toHaveBeenCalledTimes(1)
    expect(updatePendingTransactionsMock).toHaveBeenCalledWith({
      type: 'add',
      action: 'deposit',
      symbol: receiveToken.symbol,
      chainId,
    })
    expect(logTransactionArguments).toHaveBeenCalledTimes(1)
    expect(logTransactionArguments.mock.calls[0]).toMatchSnapshot()
    expect(sendMock).toHaveBeenCalledTimes(1)
    expect(sendMock.mock.calls[0]).toMatchSnapshot()
  })

  it('should process deposits with custom slippage, custom cooldown and native token', async () => {
    const depositMethod = 'depositWithCustomCooldown'
    const sendToken = {
      symbol: 'ETH',
      decimals: 18,
      address: AddressZero,
      value: '5',
      isLoading: false,
    }
    const receiveToken = {
      symbol: 'USDy',
      address: '0x123',
      value: '10',
    } as unknown as DynamicTradingToken
    const slippage = 0.15
    const tradingParams = {
      poolDepositAddress: BRIDGED_USDC_OPTIMISM.address,
      receiveAssetAddress: receiveToken.address,
      sendAssetAddress: sendToken.address,
      receiveAssetInputValue: receiveToken.value,
      fromTokenAmount: new BigNumber(sendToken.value),
    }
    vi.mocked(stateHooks.useTradingPanelDepositMethod).mockImplementation(
      () => [depositMethod, vi.fn()],
    )
    vi.mocked(stateHooks.useSendTokenInput).mockImplementation(() => [
      sendToken,
      vi.fn(),
    ])
    vi.mocked(stateHooks.useReceiveTokenInput).mockImplementation(() => [
      receiveToken,
      vi.fn(),
    ])
    vi.mocked(stateHooks.useTradingPanelSettings).mockImplementation(
      () =>
        [
          {
            slippage,
          },
          vi.fn,
        ] as unknown as ReturnType<typeof stateHooks.useTradingPanelSettings>,
    )
    vi.mocked(useTradingParams).mockImplementation(() => tradingParams)

    const { result } = renderHook(() => useDeposit())

    expect(useDepositSlippage).toHaveBeenCalledTimes(1)
    expect(useDepositSlippage).toHaveBeenCalledWith(
      tradingParams.receiveAssetInputValue,
    )
    expect(useTradingSettleHandler).toHaveBeenCalledTimes(1)
    expect(useTradingSettleHandler).toHaveBeenCalledWith('deposit')

    expect(useContractFunction).toHaveBeenCalledTimes(1)
    expect(useContractFunction).toHaveBeenCalledWith({
      contractId: 'easySwapper',
      functionName: 'depositNativeWithCustomCooldown',
      onSettled: onSettledMock,
    })

    await act(async () => await result.current())

    expect(logTransactionArguments).toHaveBeenCalledTimes(1)
    expect(logTransactionArguments.mock.calls[0]).toMatchSnapshot()
    expect(sendMock).toHaveBeenCalledTimes(1)
    expect(sendMock.mock.calls[0]).toMatchSnapshot()
  })
})
