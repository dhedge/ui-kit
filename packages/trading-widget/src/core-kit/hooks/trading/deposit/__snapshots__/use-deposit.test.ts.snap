// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useDeposit > should process deposits with auto slippage and not native token 1`] = `
[
  BuyingWithEasyswapperArgs {
    "fromTokenAmount": "5000000",
    "poolDepositAddress": "0x111",
    "receiveAssetAddress": "0x123",
    "receiveAssetInputValue": "10",
    "sendAssetAddress": "0x7F5c764cBc14f9669B88837ca1490cCa17c31607",
  },
]
`;

exports[`useDeposit > should process deposits with auto slippage and not native token 2`] = `
[
  "0x123",
  "0x7F5c764cBc14f9669B88837ca1490cCa17c31607",
  "5000000",
  "0x111",
  "10000000000000000000",
]
`;

exports[`useDeposit > should process deposits with custom slippage, custom cooldown and native token 1`] = `
[
  BuyingWithNativeAssetArgs {
    "fromTokenAmount": "5000000000000000000",
    "poolDepositAddress": "0x7F5c764cBc14f9669B88837ca1490cCa17c31607",
    "receiveAssetAddress": "0x123",
    "receiveAssetInputValue": "10",
  },
]
`;

exports[`useDeposit > should process deposits with custom slippage, custom cooldown and native token 2`] = `
[
  "0x123",
  "0x7F5c764cBc14f9669B88837ca1490cCa17c31607",
  "9985000000000000000",
  {
    "value": 5000000000000000000n,
  },
]
`;
